src/coll/mod.rs

```diff
@@ -116,19 +114,23 @@ impl Collection {
                 spec = merge_options(spec, aggregate_options);
             }
             None => {
-                let cursor = doc! { "batchSize" => (::cursor::DEFAULT_BATCH_SIZE) };
```

Comment:
This is the only change that matters; the server does not treat an explicit batch size of 0 on aggregation commands as meaning "use the server's default batch size", hence its removal.